bgs/bgs.go

```diff
@@ -992,7 +993,33 @@ func (s *BGS) createExternalUser(ctx context.Context, did string) (*models.Actor
 	}
 
 	if err := s.db.Create(&u).Error; err != nil {
-		// some debugging...
+		// If the new user's handle conflicts with an existing user,
+		// since we just validated the handle for this user, we'll assume
+		// the existing user no longer has control of the handle
+		if errors.Is(err, gorm.ErrDuplicatedKey) {
+			// Get the UID of the existing user
+			var existingUser User
+			if err := s.db.Find(&existingUser, "handle = ?", handle).Error; err != nil {
+				return nil, fmt.Errorf("failed to find existing user: %w", err)
+			}
+
+			// Set the existing user's handle to NULL and set the valid_handle flag to false
+			if err := s.db.Model(User{}).Where("id = ?", existingUser.ID).Update("handle", nil).Update("valid_handle", false).Error; err != nil {
+				return nil, fmt.Errorf("failed to update outdated user's handle: %w", err)
+			}
+
+			// Do the same thing for the ActorInfo if it exists
+			if err := s.db.Model(models.ActorInfo{}).Where("uid = ?", existingUser.ID).Update("handle", nil).Update("valid_handle", false).Error; err != nil {
+				if !errors.Is(err, gorm.ErrRecordNotFound) {
+					return nil, fmt.Errorf("failed to update outdated actorInfo's handle: %w", err)
+				}
+			}
+
+			// Create the new user
+			if err := s.db.Create(&u).Error; err != nil {
+				return nil, fmt.Errorf("failed to create user after handle conflict: %w", err)
+			}
+		}
```

Comment:
this will still return this error btw

Reply:
It's fine if we return an error here cause that means something went really wrong, right?

Reply:
That's attempting to create the user again after we've hopefully resolved the conflict

Reply:
no, what im saying is if we succeed in creating the user here, we still return an error on line 1023

Reply:
Ohhh, sorry I was reading the wrong line of code, resolved